pkg/controller/clusters/provider.go

```diff
@@ -0,0 +1,30 @@
+package clusters
+
+import (
+	"context"
+
+	eksdv1alpha1 "github.com/aws/eks-distro-build-tooling/release/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ProviderClusterReconciler struct {
+	providerClient client.Client
+}
+
+func NewProviderClusterReconciler(client client.Client) *ProviderClusterReconciler {
+	return &ProviderClusterReconciler{
+		providerClient: client,
+	}
+}
+
+func (p *ProviderClusterReconciler) GetEksdRelease(ctx context.Context, name, namespace string) (*eksdv1alpha1.Release, error) {
```

Comment:
How is this a provider specific method? Maybe I am a bit confused by the interface.

Reply:
It is not. I'm not sure these method should be part of the `ProviderClusterReconciler`. But I'm just moving code around.

I think the idea was that `ProviderClusterReconciler` was gonna be embed by all provider cluster reconciler, so we could implement common functionality in it. We might want to revisit that idea as we start using this more and probably create less generic entities.